src/client_side.cc

```diff
@@ -1638,7 +1638,10 @@ clientProcessRequest(ConnStateData *conn, const Http1::RequestParserPointer &hp,
 #endif
 
     if (internalCheck(request->url.path())) {
-        if (internalHostnameIs(request->url.host()) && request->url.port() == getMyPort()) {
+        const bool allowLocalhost = (conn->port->s.isLocalhost() || conn->port->s.isAnyAddr());
+        const bool isMyHostname = (allowLocalhost && strcmp(request->url.host(), "localhost") == 0) ||
```

Comment:
If we can rely on url.host() being lower-case, please disclose that invariant in the Uri::host() description. Otherwise, please use strcasecmp().

Reply:
Looking into this, parsed URLs enforce lower case hostnames. But the `host(...)` setter method does not  - so `append_domain` at least allowing any string input means its case cannot be trusted.
Changing to `srcasecmp()` as requested.